services/open-vsx/open-vsx-base.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService, NotFound } = require('..')
+
+const extensionQuerySchema = Joi.object({
+  error: Joi.string(),
+  version: Joi.string().when('error', {
+    is: Joi.exist(),
+    then: Joi.forbidden(),
+    otherwise: Joi.required(),
+  }),
+  timestamp: Joi.string().isoDate().when('error', {
+    is: Joi.exist(),
+    then: Joi.forbidden(),
+    otherwise: Joi.required(),
+  }),
```

Comment:
Yes, it seems that way from the API documentation.

![Open VSX Status Code](https://user-images.githubusercontent.com/1383932/98489533-f5bde600-2260-11eb-809d-6594bcc01d2c.png)

Reply:
that's unfortunate but makes sense. could we tweak this a bit to prevent potential future issues of OpenVSX ever updates their contract (for example if they start including `timestamp` even when there's an error)? Maybe even `then: Joi.optional()` instead of forbidden, though I'm sure with `Joi` there's a lot of ways of expressing that we definitely want those fields to be required when `error` is not present

Reply:
OK. I will update so that `then: Joi.forbidden()` becomes `then: Joi.optional()`.

Reply:
We are going to fix the status codes at some point. I already did that for most endpoints (https://github.com/eclipse/openvsx/issues/163), but left a few where a change would have affected existing consumers. I can notify you when we update the remaining endpoints.

Reply:
That'd be fantastic @spoenemann, thank you!